internal/ingress/controller/parser/parser.go

```diff
@@ -590,6 +623,70 @@ func overrideRoute(route *Route,
 	}
 }
 
+// normalizeProtocols prevents users from mismatching grpc/http
+func normalizeProtocols(route *Route) {
+	protocols := route.Protocols
+	var http, grpc bool
+
+	for _, protocol := range protocols {
+		if strings.Contains(*protocol, "grpc") {
+			grpc = true
+		}
+		if strings.Contains(*protocol, "http") {
+			http = true
+		}
+		if sanitizeProtocol(*protocol) != true {
+			http = true
+		}
+	}
+
+	if grpc && http {
+		route.Protocols = kong.StringSlice("http", "https")
+	}
+	// length check
+	if len(route.Protocols) < 1 {
+		return
+	}
+}
+
+// sanitizeProtocol returns a bool of whether string is a valid protocol
+func sanitizeProtocol(protocol string) bool {
+	re2 := regexp.MustCompile(`\Ahttps$|\Ahttp$|\Agrpc$|\Agrpcs$`)
```

Comment:
We don't need to compile this regex on every single call.
You can define this outside the function and then just verify each time it is called.

Reply:
as a global?

Reply:
Yes, because compiling the regex in each invocation doesn't really buy us anything.

Reply:
👍